shared/libraries/native_streaming_protocol/src/base_session_handler.cpp

```diff
@@ -358,7 +363,12 @@ void BaseSessionHandler::sendPacketBuffer(PacketBufferPtr&& packetBuffer)
     auto writeHeaderTask = createWriteHeaderTask(PayloadType::PAYLOAD_TYPE_STREAMING_PACKET, payloadSize);
     tasks.insert(tasks.begin(), writeHeaderTask);
 
-    session->scheduleWrite(std::move(tasks));
+    auto deadlineTime =
+        packetBuffer->timeStamp.has_value() && streamingPacketSendTimeout != std::chrono::milliseconds(0)
```

Comment:
if streamingPacketSendTimeout == 0, will packet buffer contain the timestamp?

Reply:
Currently, one of the two conditions is always redundant, as both are true when the feature is enabled and false when it’s disabled. 
However, since both conditions must be met to correctly set a deadline, I would keep both for now. This ensures compatibility if changes are made, such as if certain packet types, like event packets, will not have an attached timestamp.